### PR TITLE
[BUG][CM][668] | Bipul | Fix Expiry Bug on Approve and Deny API

### PR DESCRIPTION
**Pull request by bipul-jha**

### CODE REVIEW

File: src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java [hunk 1]
```diff
@@ -246,6 +249,15 @@ private static boolean isSameRequester(ConsentArtefact consentDetail, String req
...
     }
 
+    private Mono<Void> processExpiredConsentRequest(String requestId) {
+        return consentRequestRepository.requestOf(requestId)
+                .filter(consentRequest -> isConsentRequestExpired(consentRequest.getCreatedAt()) &&
+                consentRequest.getStatus().equals(REQUESTED))
+                .switchIfEmpty(Mono.empty())
+                .flatMap(response -> consentRequestRepository.updateStatus(requestId, EXPIRED).
+                        then(Mono.error(ClientError.consentRequestExpired())));
+    }
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java:259]: send notification to HIU?

there's a queue which is there for that, you may need to enqueue the message though

Comment (bipul-jha) [reply src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java:259]: Looking into it, sure.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5909948&bid=19360005)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5909948&bid=19360005)

**Pull request merged**